dspace-server-webapp/src/main/java/org/dspace/app/rest/exception/PasswordNotValidException.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.exception;
+
+import org.dspace.core.I18nUtil;
+
+/**
+ * This class provides an exception to be used when trying to create an EPerson
+ * with password that not match regular expression configured in this
+ * variable "authentication-password.regex-validation.pattern" in dspace.cfg or during the patch of password.
+ *
+ * @author Mykhaylo Boychuk (mykhaylo.boychuk@4science.com)
+ */
+public class PasswordNotValidException extends UnprocessableEntityException implements TranslatableException {
```

Comment:
@Micheleboychuk and @corrad82-4s : After testing this again, I realized that _because this is a 422 exception_, this means that any invalid passwords will have the **full stacktrace** added to the logs.

I wonder if we should consider changing this into either a `409 Conflict` or a `403 Forbidden` error?   I understand why you chose 422. But, currently we log a full stacktrace for every 422 error code...and it seems useless to log every attempt at changing a password which was deemed invalid.  
* Arguments for using 409: https://stackoverflow.com/questions/48803290/what-is-the-best-http-status-code-error-for-failed-password-update
* Arguments for using 403: https://stackoverflow.com/questions/59440865/http-status-code-for-your-password-did-not-meet-requirements

If you'd rather keep it as a 422, we can do so. But, then we may want to see if we can block this specific `PasswordNotValidException` from logging a full stacktrace.  Currently, it's VERY NOISY in logs.  Try it out, and you'll see that each "New password is invalid..." error adds about 50+ lines to the `dspace.log`.  That's a lot of useless info.

Reply:
HI @tdonohue I see your point. We have checked both logs, but honestly we are not fully convinced about using 409 or 403 as error codes for invalid password. So we'd prefer leaving 422. 
Regarding 422 noisy log, I totally agree with you, but instead of finding a workaround just for this exception, could it be worth checking if full stack trace is still needed for 422 errors? It can occur in several points of the application, there could be other exceptions logged many times with their full stack trace.

Reply:
@corrad82-4s : I think it'd be difficult for us to disable logging of all 422 errors. They include severe client errors like improper deletion, failed uploads, invalid relationships, invalid objects, etc.  Take a look at some of the 422 errors here: https://github.com/DSpace/RestContract/search?q=422

This was the main reason we added 422 to the list of default exceptions which result in a stacktrace.  Usually a 422 is a pretty severe error where an exception stacktrace is useful (to help debug if there's a larger problem).  

In the case of a "invalid" new password, the stacktrace is obviously unnecessary.  Which is why I started to question whether this error code was correct.... This exception is not nearly as severe as most other 422 exceptions.  I understand why you chose 422 though. 

I honestly don't have a better idea here.  I guess we could leave these really long stacktraces in place for now, but I expect people will complain.  Logging >50 lines every time someone tries to choose an invalid password seems like it'll get very annoying very quickly.  Disabling 422 errors from logging is another option, I guess (it'd be easy to [do here](https://github.com/DSpace/DSpace/blob/main/dspace-server-webapp/src/main/java/org/dspace/app/rest/exception/DSpaceApiExceptionControllerAdvice.java#L64)), but as I noted I suspect there are many 422 errors where detailed logging would be helpful.

Reply:
@tdonohue : thank you for your clarification. And I recall many times when 422 stack trace has been helpful in debugging. Weak password is not one of these, as you stated. 
I understand that 422 stack trace is not necessary at all, in our case, but I have some doubts in blocking stack trace logging in case of `PasswordNotValidException` , as I see it as adding kind of ad hoc behavior different than standard. 
Maybe we could leave this first version as is, then think about a logic that does not print stack trace in case of particular exceptions: stack trace depends on error code and exception.

Reply:
@corrad82-4s : That's reasonable. We can leave this as-is for now.  I do suspect though that we'll need to revisit this decision later, as I think this is one of the first scenarios where the 422 is _less severe_.  But you are correct, there may eventually be other future examples of a less severe 422 which also do not require stacktraces.